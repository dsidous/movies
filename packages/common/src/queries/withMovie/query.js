import gql from 'graphql-tag';

export const query = gql`
  query getMovie($movieId: ID!) {
    movie(id: $movieId) {
      id
      backdrop_path
      poster_path
      title
      release_date
      runtime
      vote_average
      tagline
      overview
      genres {
        id
        name
      }
      credits {
        crew {
          credit_id
          job
          name
          profile_path
        }
        cast {
          id
          credit_id
          name
          profile_path
          character
        }
      }
      reviews {
        id
        author
        content
      }
      similar {
        results {
          id
          title
          poster_path
        }
      }
      images {
        posters {
          file_path
        }
        backdrops {
          file_path
        }
      }
      videos {
        key
        type
      }
    }
  }
`;
