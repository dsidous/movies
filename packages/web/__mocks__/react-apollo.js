const apollo = require.requireActual('react-apollo');
const result = jest.fn(Component => Component);

apollo.graphql = jest.fn().mockReturnValue(result);

module.exports = apollo;
