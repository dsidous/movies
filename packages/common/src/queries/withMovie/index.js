import { graphql } from 'react-apollo';
import { query } from './query';

const withMovie = () =>
  graphql(query, {
    options: props => ({
      fetchPolicy: 'cache-first',
      variables: {
        movieId: props.movieId,
      },
    }),
    props: ({ data: { movie = {}, loading } }) => ({
      loading,
      show: movie,
    }),
  });

export default withMovie;
