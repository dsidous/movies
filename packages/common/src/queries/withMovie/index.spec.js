import { graphql } from 'react-apollo';
import React from 'react';
import { shallow } from 'enzyme';
import withMovie from '.';

jest.mock('react-apollo', () => ({
  ...jest.requireActual('react-apollo'),
  graphql: jest.fn(),
}));

describe('Queries/withMovie', () => {
  graphql.mockImplementation((_, config) => {
    const ownProps = { movieId: 1 };
    const props = config.props({
      data: {
        movie: {
          id: 1,
        },
        loading: true,
      },
      ownProps,
    });
    const options = config.options(ownProps);
    return jest.fn(Mock => <Mock options={options} {...props} />);
  });

  const MockComponent = props => ({ ...props });
  const Component = withMovie()(MockComponent);

  it('should render as expected', () => {
    const render = shallow(Component);
    expect(render).toMatchSnapshot();
  });
});
