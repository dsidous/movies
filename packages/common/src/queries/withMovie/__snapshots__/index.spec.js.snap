// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Queries/withMovie should render as expected 1`] = `ShallowWrapper {}`;
